unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/LOD.cs

```diff
@@ -0,0 +1,231 @@
+                visibility.SetCombinedRendererVisibility(lodIndex <= 1);
+                visibility.SetFacialFeaturesVisibility(lodIndex <= 0);
+                SetImpostorEnabled(avatarAlpha == 0);
+                return;
+            }
+
+            if (lodIndex == this.lodIndex)
+                return;
+
+            this.lodIndex = lodIndex;
+
+            transitionCTS?.Cancel();
+            transitionCTS?.Dispose();
+            transitionCTS = new CancellationTokenSource();
+            Transition(transitionCTS.Token);
+        }
+
+        private Renderer CreateImpostor()
+        {
+            GameObject quadImpostorContainer = Object.Instantiate(Resources.Load<GameObject>("QuadImpostor"), impostorContainer.transform);
+            return quadImpostorContainer.GetComponent<Renderer>();
+        }
+
+        private async UniTaskVoid Transition(CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            try
+            {
+                visibility.SetCombinedRendererVisibility(true);
+                impostorRenderer.enabled = true;
+                float targetAvatarAlpha = lodIndex <= 1 ? 1f : 0f;
+                while (!Mathf.Approximately(targetAvatarAlpha, avatarAlpha))
+                {
+                    avatarAlpha = Mathf.MoveTowards(avatarAlpha, targetAvatarAlpha, (1f / TRANSITION_DURATION) * Time.deltaTime);
+                    UpdateAlpha(avatarAlpha);
+                    await UniTask.NextFrame(ct);
+                }
+
+                UpdateSSAO(combinedAvatar, lodIndex);
+                UpdateMovementLerping(lodIndex);
+
+                visibility.SetCombinedRendererVisibility(avatarAlpha > 0);
+                visibility.SetFacialFeaturesVisibility(lodIndex == 0);
+
+                SetImpostorEnabled(avatarAlpha == 0);
+            }
+            catch (OperationCanceledException)
+            {
+                //No disposing required
+                throw;
+            }
+        }
+
+        private void SetImpostorEnabled(bool enabled)
+        {
+            impostorRenderer.enabled = enabled;
+            billboardLookAtCameraCTS?.Cancel();
+            billboardLookAtCameraCTS?.Dispose();
+            if (enabled)
+            {
+                billboardLookAtCameraCTS = new CancellationTokenSource();
+                BillboardLookAtCamera(billboardLookAtCameraCTS.Token);
+            }
+            else
+            {
+                billboardLookAtCameraCTS = null;
+            }
+        }
+
+        internal void UpdateAlpha(float avatarAlpha)
+        {
+            if (combinedAvatar == null)
+                return;
+
+            Material[] mats = combinedAvatar.sharedMaterials;
+            for (int j = 0; j < mats.Length; j++)
+            {
+                mats[j].SetFloat(AvatarSystemUtils.DitherFade, avatarAlpha);
+            }
+
+            Material impostorMaterial = impostorRenderer.material;
```

Comment:
Using material and not sharedMaterial here was intended? If so let's add a comment please.

Reply:
In our previous `AvatarRenderer` we were using `material` I think it is intended, @pravusjif do you have any insight on this?